lib/dynamoid/persistence.rb

```diff
@@ -145,6 +147,22 @@ def dynamo_type(type)
         end
       end
 
+      private
+
+      def transform_hash(hash)
+        {}.tap do |h|
+          hash.each { |key, value| h[key.to_sym] = transform(value) }
+        end
+      end
+
+      def transform(val)
+        case val
+        when BigDecimal; val.to_f
+        when Hash; transform_hash(val)
+        when Array; val.map { |v| transform(v) }
+        else; val
```

Comment:
Stylistically, this repo seems to prefer multiline `case/when` instead of `;` or `then`:
```
case
...
when Array
  val.map { |v| transform(v) }
else
  val
end
```